### PR TITLE
[fix](priv) fix bug of grant priv on ctl.db.* not work

### PR DESCRIPTION
**Pull request by Yulei-Yang**
currently, when use grant xxx_priv on ctl.db.* to user_a, it does not work. When user_a switch to ctl, he cannot see or use any database.

# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [x] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yulei-Yang** [comment]
./run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/DbPrivTable.java [hunk 1]
```diff
@@ -42,7 +45,8 @@ public void getPrivs(String ctl, String db, PrivBitSet savedPrivs) {
             }
 
             // check db
-            if (!dbPrivEntry.isAnyDb() && !dbPrivEntry.getDbPattern().match(db)) {
+            if (!dbPrivEntry.isAnyDb() && !dbPrivEntry.getDbPattern().match(db) && !dbPrivEntry.getDbPattern()
```
Comment (lide-reed) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/DbPrivTable.java:48]: Can you add a comment

Comment (Yulei-Yang) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/DbPrivTable.java:49]: fixed

**Comment by Yulei-Yang** [comment]
run p0

**Review by lide-reed — APPROVED** [review]
LGTM

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Comment by Yulei-Yang** [comment]
run p0

**Review by morningman — APPROVED** [review]
LGTM

**Comment by Yulei-Yang** [comment]
run beut

**Comment by Yulei-Yang** [comment]
run feut

**Comment by Yulei-Yang** [comment]
run feut

**Comment by Yulei-Yang** [comment]
run feut

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**